src/transformers/data/datasets/squad.py

```diff
@@ -128,28 +129,35 @@ def __init__(
         with FileLock(lock_path):
             if os.path.exists(cached_features_file) and not args.overwrite_cache:
                 start = time.time()
-                self.features = torch.load(cached_features_file)
+                cache_dict = torch.load(cached_features_file)
+                self.features = cache_dict["features"]
```

Comment:
Here and below, use `.get(, None)` so as to not break with earlier cached directories. Then we can issue a warning to clean the cache dir if one of those attrs is `None` saying a new version of the preprocessing is availble.

Reply:
Fixed. Added a comment and warning with regard to support for the legacy cache file